pkg/controller/certificates/issuing/issuing_controller.go

```diff
@@ -178,6 +178,17 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 		return err
 	}
 
+	// Make sure there isn't another Certificate with the same secret name in the same namespace.
+	certs, err := c.certificateLister.Certificates(namespace).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	for _, ct := range certs {
+		if crt.Spec.SecretName == ct.Spec.SecretName && crt.ObjectMeta.Name != ct.ObjectMeta.Name {
+			return nil
+		}
+	}
+
```

Comment:
I'm not necessarily proposing we copy nearly identical code into these three different controllers. I just wasn't sure where it could go otherwise; hoping to discuss a little.

Unfortunately, I'm not able to give y'all edit permissions on our fork of this repository.